[Vitest Integration] document known issues with isolated storage (#20033)
cloudflare/cloudflare-docs: src/content/docs/workers/testing/vitest-integration/known-issues.mdx

Durable Object alarms are not reset between test runs and do not respect isolated storage. Ensure you delete or run all alarms with [`runDurableObjectAlarm()`](/workers/testing/vitest-integration/test-apis/#durable-objects) scheduled in each test before finishing the test.

### WebSockets

Using WebSockets with Durable Objects with the [`isolatedStorage`](/workers/testing/vitest-integration/isolation-and-concurrency) flag turned on is not supported. You must set `isolatedStorage: false` in your `vitest.config.ts` file.

### Isolated storage

> When the `isolatedStorage` flag is enabled (the default setting), follow these patterns to avoid issues:

Reviewer: This could maybe do with a link to the isolated storage docs, and some discussion of the tradeoffs of disabling it?

> - Await All Storage Operations

Reviewer: This shouldn't be title case

> Always `await` all `Promise`s that read or write to storage services.

Reviewer: This should have an example, I think?

- Explicit Resource Disposal

> If the RPC methods return non-primitive values (such as objects or classes extending `RpcTarget`), use the `using` keyword to explicitly signal when resources can be disposed. See [this example test](https://github.com/cloudflare/workers-sdk/tree/main/fixtures/vitest-pool-workers-examples/rpc/test/unit.test.ts#L155) and refer to [explicit-resource-management](/workers/runtime-apis/rpc/lifecycle#explicit-resource-management) for more details.

Reviewer: What are "the" RPC methods in this context?

Durable Object alarms are not reset between test runs and do not respect isolated storage. Ensure you delete or run all alarms with [`runDurableObjectAlarm()`](/workers/testing/vitest-integration/test-apis/#durable-objects) scheduled in each test before finishing the test.

### WebSockets

Using WebSockets with Durable Objects with the [`isolatedStorage`](/workers/testing/vitest-integration/isolation-and-concurrency) flag turned on is not supported. You must set `isolatedStorage: false` in your `vitest.config.ts` file.

### Isolated storage

> When the `isolatedStorage` flag is enabled (the default), the test runner will undo any writes to the storage at the end of the test as detailed in the [Isolation and concurrency docs](/workers/testing/vitest-integration/isolation-and-concurrency/). However, you must manage resources carefully. Below are some common pitfalls and how to avoid them:

Reviewer 2: ```suggestion
When the `isolatedStorage` flag is enabled (the default), the test runner will undo any writes to the storage at the end of the test as detailed in the [isolation and concurrency documentation](/workers/testing/vitest-integration/isolation-and-concurrency/). However, Cloudflare recommends that you consider the following actions to avoid any common issues:
```
Just trying to make it sound a little more formal and authoritative

> - Await all storage operations

Reviewer 2: ```suggestion
## Await all storage operations
```

Always `await` all `Promise`s that read or write to storage services.

```ts
// Example: Seed data
beforeAll(async () => {
	await env.KV.put('message', 'test message');
	await env.R2.put('file', 'hello-world');
});
```

> - Explicit resource disposal

Reviewer 2: ```suggestion
## Explicitly signal resource disposal
```
Making all these suggestions all verb-driven gives them parallel structure

> When calling RPC methods of a Service Worker or Durable Object that return non-primitive values (such as objects or classes extending `RpcTarget`), use the `using` keyword to explicitly signal when resources can be disposed. See [this example test](https://github.com/cloudflare/workers-sdk/tree/main/fixtures/vitest-pool-workers-examples/rpc/test/unit.test.ts#L155) and refer to [explicit-resource-management](/workers/runtime-apis/rpc/lifecycle#explicit-resource-management) for more details.

Reviewer 2: ```suggestion
When calling RPC methods of a Service Worker or Durable Object that return non-primitive values (such as objects or classes extending `RpcTarget`), use the `using` keyword to explicitly signal when resources can be disposed of. See [this example test](https://github.com/cloudflare/workers-sdk/tree/main/fixtures/vitest-pool-workers-examples/rpc/test/unit.test.ts#L155) and refer to [explicit-resource-management](/workers/runtime-apis/rpc/lifecycle#explicit-resource-management) for more details.
```
Please check me on this one.

```ts
using result = await stub.getCounter();
```

> - Consume response bodies

Reviewer 2: ```suggestion
## Consume response bodies
```

> When making requests via `fetch` or `R2.get()`, consume the entire response body, even if you aren't asserting its content. For example:

Reviewer 2: ```suggestion
When making requests via `fetch` or `R2.get()`, consume the entire response body, even if you are not asserting its content. For example:
```
